src/components/DatePicker/CalendarDialog.tsx

```diff
@@ -65,6 +64,7 @@ interface CalendarDialogProps {
   date: Date | null
   title?: string
   timeFrame?: TimeFrame
+  dateRange?: { start: Date; end: Date }
```

Comment:
**suggestion:** What do you think about making this required if the timeframe is custom?

Something like this:

```ts
interface DatePickerProps {
  date?: Date | null
  placeholder: string
  title?: string
  info?: string
  monthLabel: string
  yearLabel: string
  locale?: string
  onChange: (date: Date) => void
}

type TimeFrameProps =
  | { timeFrame?: never; dateRange?: never }
  | { timeFrame?: TimeFrame.all; dateRange?: never }
  | { timeFrame?: TimeFrame.past; dateRange?: never }
  | { timeFrame?: TimeFrame.future; dateRange?: never }
  | { timeFrame?: TimeFrame.custom; dateRange: { start: Date; end: Date } }

...

export const DatePicker = ({
  date = null,
  placeholder,
  title,
  timeFrame = TimeFrame.future,
  dateRange,
  info,
  monthLabel,
  yearLabel,
  locale = 'en-US',
  onChange,
}: DatePickerProps & TimeFrameProps)
```

Reply:
For some reason, Typescript doesn't recognize the enums as being part of TimeFrame. Let's move on without this for now, and revisit later.